doc/lua_api.md

```diff
@@ -8527,6 +8529,12 @@ Player properties need to be saved manually.
     show_on_minimap = false,
     -- Defaults to true for players, false for other entities.
     -- If set to true the entity will show as a marker on the minimap.
+
+    engine_mask = 7,
+    -- This can be used for enable/disable some part of engine.
+    -- 0x01 - Drowning (Only avaliable for players)
+    -- 0x02 - Breathing (Only avaliable for players)
+    -- 0x04 - Node damage
```

Comment:
Bitmasks seem like a foreign thing in a simple language like Lua. I think we don't have a bitmask anywhere else in our Lua API.

Reply:
We have bitsets in a couple places, the `style` property of HUD `text` elements for example ("Bitfield with 1 = bold, 2 = italic, 4 = monospace"), but I agree that this should ideally be hidden from the modder: A Lua-friendly API should simply provide boolean fields unless there is a very strong reason not to (like performance when considering `param2` in nodes for example), which I don't see here - object properties could still use a bitset under the hood. That said

> Maybe it would be better to keep this PR simple and uncontroversial by just including the "allow disabling hardcoded engine features" change.

Yes, I would prefer this. I think it's the best idea to leave the (re)implementation of these features for entities up to modders.